boards/shields/arduino_uno_click/boards/nrf9160dk_nrf9160_common.dtsi

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2024 Golioth, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&pinctrl {
+	/*
+	 * The original Arduino Uno provides the same SCL/SDA on two sets of
+	 * pins, but the nRF9160 DK maps these pins to two different pairs of
+	 * GPIO. When using the Arduino Uno Click Shield board with the nRF9160
+	 * DK, the P0.18/P0.19 pair must be used.
+	 */
```

Comment:
Genuine question here: If this description is indeed correct, why not fixing the arduino_i2c configuration in board directly and point it to another i2c instance with a pin configuration matching arduino configuration ?
https://github.com/zephyrproject-rtos/zephyr/blob/98d0fe4452a2042bf8daf28654ca902bae964fcd/boards/nordic/nrf9160dk/nrf9160dk_nrf9160_common.dtsi#L195

Reply:
I don't believe the existing `arduino_i2c` configuration in the nRF9160 DK board is wrong, so I'm not sure that the best option would be to change it. Let me explain.

The [Arduino Uno schematic](https://www.arduino.cc/en/uploads/Main/Arduino_Uno_Rev3-schematic.pdf) shows that the same I2C pins are routed to two different headers on the board:

<img width="689" alt="image" src="https://github.com/zephyrproject-rtos/zephyr/assets/14631/63cd15da-22e0-4c44-af76-a80276fd1e65">

As a result, an Arduino Shield can technically choose to use either set of header pins for I2C and they will be connected to the same I2C pins on the Arduino MCU.

However, from the Arduino Uno pinout diagram, it appears the "recommended" pins for shields to use for I2C are the `D18`/`D19` pins:

<img width="930" alt="image" src="https://github.com/zephyrproject-rtos/zephyr/assets/14631/cb552a87-8a91-4a1d-8c6f-c5d477436756">

The nRF9160 DK board follows the recommendation in the Arduino Uno pinout diagram (i.e. `arduino_i2c` is connected to the recommended `SCL`/`SDA` (`D18`/`D19`) pins in the pinout diagram).

<img width="613" alt="image" src="https://github.com/zephyrproject-rtos/zephyr/assets/14631/3cadf0c3-1046-4f93-b074-17d34da45a9f">

**To me, it looks like the existing pinctrl configuration for the nRF9160 DK board is correct because the majority of Arduino shields will use the `D18`/`D19` pins recommended in the pinout diagram.**

However, in the specific combination of the nRF9160 DK + Arduino Uno Click shield, there are two things that cause an incompatibility:

1. The nRF9160 DK routes two different sets of nRF9160 SIP pins to the Arduino I2C pins on the headers:
   - `P0.18`/`P0.19`: Arduino `AD4`/`AD5`
   - `P0.30`/`P0.31`: Arduino `D18`/`D19`
2. The Arduino Uno Click shield only routes the mikroBUS I2C pins to the `AD4`/`AD5` (`P0.18`/`P0.19`) pins on the shield:

<img width="845" alt="image" src="https://github.com/zephyrproject-rtos/zephyr/assets/14631/67d4c9f5-231b-40d0-8d2b-983e5205b452">

**The end result is that with this specific combination of boards, the pins used for `arduino_i2c` by default (`P0.30`/`P0.31`) are not connected to anything on the Arduino Uno Click shield.**

> why not fixing the arduino_i2c configuration in board directly and point it to another i2c instance with a pin configuration matching arduino configuration ?

The nRF9160 support in Zephyr does not currently support having multiple peripherals enabled with the same [Peripheral ID](https://docs.nordicsemi.com/bundle/ps_nrf9160/page/peripheral_interface.html#ariaid-title2) simultaneously. There are 4 peripheral IDs available in the nRF9160: 0-3. 

Currently, the following interfaces are enabled by default (all 4 peripheral IDs are in use):

- ID 0: `zephyr,console` (`uart0`)
- ID 1: `arduino_serial` (`uart1`)
- ID 2: `arduino_i2c` (`i2c2`)
- ID 3: `arduino_spi` (`spi3`)

There are no remaining Peripheral IDs to use for enabling another I2C interface in addition to `i2c2`.

If we wanted to "fix" the current configuration to work with the Arduino Uno Click shield, we would have to make changes to the existing enabled peripherals:

1. Disable `i2c2` on `P0.30`/`P0.31` and enable it on `P0.18`/`P0.19` (this may break existing samples/apps)
2. Disable one of the other peripherals and assign `P0.18`/`P0.19` to that peripheral ID (e.g. disable `spi3` and enable `i2c3, but this may break existing samples/apps which rely on `spi3` to be enabled).

Both of these options seem to have the potential to break existing apps, so it seemed like the 3rd option that doesn't break anything is to just add the board overlay for the shield to remap the `i2c2` pins from `P0.30`/`P0.31` → `P0.18`/`P0.19` (since `P0.30`/`P0.31` aren't usable when the shield is installed).

LMK if that logic makes sense or if I'm missing a better option for this.

Reply:
> To me, it looks like the existing pinctrl configuration for the nRF9160 DK board is correct because the majority of Arduino shields will use the D18/D19 pins recommended in the pinout diagram.

Ok, then let's keep it that way.